pygfx/utils/show.py

```diff
@@ -11,56 +14,104 @@
+            A callback that will be executed during each draw call before a new
+            render is made.
+        after_render : Callable
+            A callback that will be executed during each draw call after a new
+            render is made.
+        draw_function : Callable
+            Replaces the draw callback with a custom one. If set both
+            `before_render` and `after_render` will have no effect.
 
-    Parameters:
-        object (WorldObject): The object to show.
-        up (Vector3): Optional. Configure the up vector for the camera controller.
     """
-    from wgpu.gui.auto import WgpuCanvas, run
 
     if isinstance(object, Scene):
         scene = object
     else:
         scene = Scene()
         scene.add(object)
 
-        background = Background(None, BackgroundMaterial((0, 1, 0, 1), (0, 1, 1, 1)))
+        dark_gray = np.array((169, 167, 168, 255)) / 255
+        light_gray = np.array((100, 100, 100, 255)) / 255
+
+        background = Background(None, BackgroundMaterial(light_gray, dark_gray))
         scene.add(background)
+        scene.add(AmbientLight(0.2))
+
+    if not camera:
+        camera = PerspectiveCamera(70, 16 / 9)
+        camera.add(DirectionalLight(0.8))
+        scene.add(camera)
 
-    camera = PerspectiveCamera(70, 16 / 9)
-    look_at = camera.show_object(object)
-    scene.add(camera)
+    if renderer is not None:
+        canvas = renderer.target
+    elif canvas is None:
+        from wgpu.gui.auto import WgpuCanvas
 
-    # Add lights if there are none
-    light_count = 0
+        canvas = WgpuCanvas()
+        renderer = WgpuRenderer(canvas)
+    else:
+        renderer = WgpuRenderer(canvas)
 
-    def check_light(ob):
-        nonlocal light_count
-        if isinstance(ob, Light):
-            light_count += 1
+    if controller is None:
+        look_at = camera.show_object(object)
+        controller = OrbitController(camera.position.clone(), look_at, up=up)
+        controller.add_default_event_handlers(renderer, camera)
 
-    scene.traverse(check_light, False)
-    if not light_count:
-        camera.add(DirectionalLight(0.8))
-        scene.add(AmbientLight(0.2))
+    if draw_function is None:
+
+        def animate():
+            if before_render is not None:
+                before_render()
+
+            controller.update_camera(camera)
+            renderer.render(scene, camera)
 
-    canvas = WgpuCanvas()
-    renderer = WgpuRenderer(canvas)
+            if after_render is not None:
+                after_render()
 
-    controller = OrbitController(camera.position.clone(), look_at, up=up)
-    controller.add_default_event_handlers(renderer, camera)
+            renderer.request_draw()
 
-    def animate():
-        controller.update_camera(camera)
-        renderer.render(scene, camera)
+        draw_function = animate
 
-    canvas.request_draw(animate)
-    run()
+    canvas.request_draw(draw_function)
+    sys.modules[renderer.target.__module__].run()
```

Comment:
Use Canvas like in the line above.
```suggestion
    sys.modules[canvas.__module__].run()
```